lib/puppet/parser/functions/delete.rb

```diff
@@ -20,19 +20,28 @@ module Puppet::Parser::Functions
 
     delete('abracadabra', 'bra')
     Would return: 'acada'
+
+    delete(['abracadabra'], '^.*bra.*$', true)
```

Comment:
I don't like passing in a magic 'true' at the end of this function. That's bad ux.